src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +526,50 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Option[String] = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    // check if group exists
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    val existingGroup = Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(f) =>
+        logger.warn(s"Google Group $groupEmail does not exist")
+        "" // return nothing in case of failure
+      case Success(_) => {
+        groupEmail // return groupEmail
+      }
+    }
+    Option(existingGroup)
+  }
+
+  override def createGoogleGroup(groupEmail: String, targetUserEmail: String): Option[String] = {
+    val newGroup = new Group().setEmail(groupEmail)
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+
+    val insertRequest = directoryService.groups.insert(newGroup)
+    val setEmail = Try(executeGoogleRequest[Group](insertRequest)) match {
+      case Failure(f) => {
+        logger.warn(s"Could not create new group called $groupEmail")
+        "" // return empty string
+      }
+      case Success(newGroupInfo) => {
+        // add targetUserEmail as member of google group - modeled after `override def addMemberToGroup` in workbench-libs HttpGoogleDirectoryDAO.scala
+        val member = new Member().setEmail(targetUserEmail).setRole("MEMBER").setDeliverySettings("ALL_MAIL")
+        val memberInsertRequest = directoryService.members.insert(groupEmail, member)
+
+        Try(executeGoogleRequest(memberInsertRequest)) match {
+          case Failure(f) =>
+            logger.warn(s"Could not add new member $targetUserEmail to group $groupEmail")
+            "" // return nothing
```

Comment:
Question about this failure case. If we successfully created the group, but were not able to add the member, does that leave the group as a existing group? Would future retries with the same values then fail on the `insertRequest`?

Reply:
I think that that failure case would leave that Google group existing but empty, and not associated with the user (since it would not be set as `anonymousGroup` in Thurloe - this logic happens in  the function `callCreateGoogleGroup` in `UserService`. So the next time this sequence happens, the process would repeat with a new group. It's not the most elegant because it does leave that empty Google group but I don't think it harms anything. Very open to suggestions, though, @rushtong!

Reply:
As a generic safety check, it *should* delete the group in case of user association failure, but I don't have any real sense of how much of a problem it would be if that didn't happen. Given the nature of random names, it seems extraordinarily unlikely to hit a duplicate.

Reply:
okay, I wrote a function to do so but still need to create tests to ensure it works properly